# HG changeset patch
# User andrew
# Date 1434497865 -3600
#      Wed Jun 17 00:37:45 2015 +0100
# Node ID adf1b3700c5ba6c13eb8f9d8eb4d6d32085bd483
# Parent  9e470ada9ea747f711ad2bd5b69e8124fbad384f
PR2428: OpenJDK build can't handle commas in LDFLAGS
Summary: Expand LDFLAGS outside function call, as is done with CFLAGS

diff -r 9e470ada9ea7 -r adf1b3700c5b make/CompileDemos.gmk
--- openjdk/jdk/make/CompileDemos.gmk	Mon Jun 08 19:25:43 2015 +0100
+++ openjdk/jdk/make/CompileDemos.gmk	Wed Jun 17 00:37:45 2015 +0100
@@ -246,12 +246,13 @@
       -I$(JDK_TOPDIR)/src/share/demo/jvmti/$1 $$(BUILD_DEMO_JVMTI_$1_EXTRA_INC) $3
 
   # Remove the -incremental:no setting to get .ilk-files like in the old build.
+  BUILD_DEMO_JVMTI_$1_LDFLAGS := $(filter-out -incremental:no -opt:ref, $(LDFLAGS_JDKLIB))
+  
   $$(eval $$(call SetupNativeCompilation,BUILD_DEMO_JVMTI_$1, \
       SRC := $(JDK_TOPDIR)/src/share/demo/jvmti/$1 $$(BUILD_DEMO_JVMTI_$1_EXTRA_SRC), \
       LANG := $$(BUILD_DEMO_JVMTI_$1_LANG), \
       OPTIMIZATION := LOW, \
       CXXFLAGS := $$($1_CXXFLAGS), \
-      LDFLAGS := $(filter-out -incremental:no -opt:ref, $(LDFLAGS_JDKLIB)), \
       LDFLAGS_macosx := $(call SET_EXECUTABLE_ORIGIN), \
       LDFLAGS_SUFFIX := $$($1_EXTRA_CXX), \
       LDFLAGS_SUFFIX_posix := $5, \
